compass_app/app/integration_test/app_server_data_test.dart

```diff
@@ -0,0 +1,207 @@
+      await tester.pumpAndSettle();
+
+      // Home screen
+      expect(find.byType(HomeScreen), findsOneWidget);
+      await tester.pumpAndSettle();
+
+      // Should show user name
+      expect(find.text('Sofie\'s Trips'), findsOneWidget);
+
+      // Tap on booking (Alaska is created by default)
+      await tester.tap(find.text('Alaska, North America'));
+      await tester.pumpAndSettle();
+
+      // Should be at booking screen
+      expect(find.byType(BookingScreen), findsOneWidget);
+      expect(find.text('Alaska'), findsOneWidget);
+
+      // Navigate back to home
+      await tester.tap(find.byType(HomeButton).first);
+      await tester.pumpAndSettle();
+
+      // Home screen
+      expect(find.byType(HomeScreen), findsOneWidget);
+
+      // Perform logout
+      await tester.tap(find.byType(LogoutButton).first);
+      await tester.pumpAndSettle();
+      expect(find.byType(LoginScreen), findsOneWidget);
+    });
+
+    testWidgets('Create booking', (tester) async {
+      // Load app widget with local configuration
+      await tester.pumpWidget(
+        MultiProvider(
+          providers: providersRemote,
+          child: const MainApp(),
+        ),
+      );
+
+      await tester.pumpAndSettle();
+
+      // Login screen because logget out
+      expect(find.byType(LoginScreen), findsOneWidget);
+
+      // Perform login (credentials are prefilled)
+      await tester.tap(find.text('Login'));
+      await tester.pumpAndSettle();
+
+      // Home screen
+      expect(find.byType(HomeScreen), findsOneWidget);
+      await tester.pumpAndSettle();
+
+      // Select create new booking
+      await tester.tap(find.byKey(const ValueKey('booking-button')));
+      await tester.pumpAndSettle();
+
+      // Search destinations screen
+      expect(find.byType(SearchFormScreen), findsOneWidget);
+
+      // Select Europe because it is always the first result
+      await tester.tap(find.text('Europe'), warnIfMissed: false);
+
+      // Select dates
+      await tester.tap(find.text('Add Dates'));
+      await tester.pumpAndSettle();
+      final tomorrow = DateTime.now().add(const Duration(days: 1)).day;
+      final nextDay = DateTime.now().add(const Duration(days: 2)).day;
+      // Select first and last widget that matches today number
+      //and tomorrow number, sort of ensures a valid range
+      await tester.tap(find.text(tomorrow.toString()).first);
+      await tester.tap(find.text(nextDay.toString()).last);
+      await tester.pumpAndSettle();
+      await tester.tap(find.text('Save'));
+      await tester.pumpAndSettle();
+
+      // Select guests
+      await tester.tap(find.byKey(const ValueKey('add_guests')),
+          warnIfMissed: false);
+
+      // Refresh screen state
+      await tester.pumpAndSettle();
+
+      // Perform search and navigate to next screen
+      await tester.tap(find.byKey(const ValueKey('submit_button')));
+      await tester.pumpAndSettle(const Duration(seconds: 2));
```

Comment:
Someplace you should document how you came up with 2 seconds (and I think the number should be much shorter for all of these). I think you should only need multi section settle if there is an animation playing.

Reply:
I think I left these by mistake when I was implementing the integration tests. After running the tests again without the duration value, it works as well, so they can be removed.

Reply:
@ericwindmill do you want me to push directly to `compass-app`, or should I create a new PR?

Reply:
@miquelbeltran I've been pushing directly when addressing these comments. Feel free to do the same